### DISCUSSION TITLE
Error when uploading a GPX track recorded by OsmAnd to OSM

### DISCUSSION THREAD

**Discussion by kreuzschnabel** [osmandapp/OsmAnd#12185] [General]
Hi,

I just tried to upload a GPX track recorded with OsmAnd to the OSM server. The track then did not process properly but was rejected, and I received the following error message which I quote in full:

> Error: Namespace prefix osmand on speed is not defined at :50. /srv/www.openstreetmap.org/rails/config/initializers/libxml.rb:4:in `block in <main>' /srv/www.openstreetmap.org/rails/lib/gpx.rb:16:in `read' /srv/www.openstreetmap.org/rails/lib/gpx.rb:16:in `parse_file' /srv/www.openstreetmap.org/rails/lib/gpx.rb:58:in `rescue in points' /srv/www.openstreetmap.org/rails/lib/gpx.rb:46:in `points' /srv/www.openstreetmap.org/rails/app/models/trace.rb:253:in `each' /srv/www.openstreetmap.org/rails/app/models/trace.rb:253:in `each_slice' /srv/www.openstreetmap.org/rails/app/models/trace.rb:253:in `import' /srv/www.openstreetmap.org/rails/app/jobs/trace_importer_job.rb:5:in `perform' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/execution.rb:48:in `block in perform_now' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:117:in `block in run_callbacks' /var/lib/gems/2.7.0/gems/i18n-1.8.10/lib/i18n.rb:314:in `with_locale' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/translation.rb:9:in `block (2 levels) in <module:Translation>' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/time/zones.rb:66:in `use_zone' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/timezones.rb:9:in `block (2 levels) in <module:Timezones>' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/instrumentation.rb:21:in `block in instrument' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/notifications.rb:203:in `block in instrument' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/notifications/instrumenter.rb:24:in `instrument' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/notifications.rb:203:in `instrument' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/instrumentation.rb:31:in `instrument' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/instrumentation.rb:14:in `block (2 levels) in <module:Instrumentation>' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/logging.rb:22:in `block in tag_logger' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/tagged_logging.rb:99:in `block in tagged' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/tagged_logging.rb:37:in `tagged' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/tagged_logging.rb:99:in `tagged' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/logging.rb:22:in `tag_logger' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/logging.rb:15:in `block (2 levels) in <module:Logging>' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:137:in `run_callbacks' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/execution.rb:47:in `perform_now' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/execution.rb:25:in `block in execute' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:117:in `block in run_callbacks' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/railtie.rb:47:in `block (4 levels) in <class:Railtie>' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/execution_wrapper.rb:88:in `wrap' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/reloader.rb:72:in `block in wrap' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/execution_wrapper.rb:88:in `wrap' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/reloader.rb:71:in `wrap' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/railtie.rb:46:in `block (3 levels) in <class:Railtie>' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks' /var/lib/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:137:in `run_callbacks' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/execution.rb:23:in `execute' /var/lib/gems/2.7.0/gems/activejob-6.1.4/lib/active_job/queue_adapters/delayed_job_adapter.rb:42:in `perform' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/backend/base.rb:81:in `block in invoke_job' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:61:in `block in initialize' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:66:in `execute' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:40:in `run_callbacks' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/backend/base.rb:78:in `invoke_job' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:230:in `block (2 levels) in run' /usr/lib/ruby/2.7.0/timeout.rb:95:in `block in timeout' /usr/lib/ruby/2.7.0/timeout.rb:105:in `timeout' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:230:in `block in run' /usr/lib/ruby/2.7.0/benchmark.rb:308:in `realtime' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:229:in `run' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:312:in `block in reserve_and_run_one_job' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:61:in `block in initialize' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:66:in `execute' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:40:in `run_callbacks' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:312:in `reserve_and_run_one_job' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:213:in `block in work_off' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:212:in `times' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:212:in `work_off' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:175:in `block (4 levels) in start' /usr/lib/ruby/2.7.0/benchmark.rb:308:in `realtime' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:174:in `block (3 levels) in start' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:61:in `block in initialize' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:66:in `execute' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:40:in `run_callbacks' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:173:in `block (2 levels) in start' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:172:in `loop' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:172:in `block in start' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/plugins/clear_locks.rb:7:in `block (2 levels) in <class:ClearLocks>' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:79:in `block (2 levels) in add' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:61:in `block in initialize' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:79:in `block in add' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:66:in `execute' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/lifecycle.rb:40:in `run_callbacks' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/worker.rb:171:in `start' /var/lib/gems/2.7.0/gems/delayed_job-4.1.9/lib/delayed/tasks.rb:9:in `block (2 levels) in <main>' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `each' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `execute' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:219:in `block in invoke_with_call_chain' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `synchronize' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/task.rb:188:in `invoke' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `each' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:125:in `run_with_threads' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:110:in `top_level' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:83:in `block in run' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling' /var/lib/gems/2.7.0/gems/rake-13.0.6/lib/rake/application.rb:80:in `run' /var/lib/gems/2.7.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>' /usr/local/bin/rake:23:in `load' /usr/local/bin/rake:23:in `<main>' 

OsmAnd Version: OsmAnd+ 3.9.10
Android 11, Lineage 18.1 latest nightly
Device model: OnePlus 5T

--ks

**Comment by vshcherb** [comment]
This looks like internal issues with OSM website, not clear what went wrong. Please try to upload that track via web interface.
